src/main/java/dev/blaauwendraad/masker/json/ValueMaskers.java

```diff
@@ -258,7 +278,182 @@ public static ValueMasker.AnyValueMasker withRawValueFunction(Function<String, S
+     *     <td>{@code value -> value.replaceAll("secret", "***")}
+     *     <td>{@code { "maskMe": "a ***" }}
+     *   <tr>
+     *     <td>{@code { "maskMe": 12345 }}
+     *     <td>{@code value -> value.startsWith("123") ? "0" : value}
+     *     <td>{@code { "maskMe": "0" }}
+     *   <tr>
+     *     <td>{@code { "maskMe": 12345 }}
+     *     <td>{@code value -> value}
+     *     <td>{@code { "maskMe": "12345" }}
+     *   <tr>
+     *     <td>{@code { "maskMe": "secret" }}
+     *     <td>{@code value -> "***"}
+     *     <td>{@code { "maskMe": "***" }}
+     *   <tr>
+     *     <td>{@code { "maskMe": "secret value" }}
+     *     <td>{@code value -> value.substring(0, 3) + "***"}
+     *     <td>{@code { "maskMe": "sec***" }}
+     *   <tr>
+     *     <td>{@code { "maskMe": "Andrii \"Juice\" Pilshchykov" }}
+     *     <td>{@code value -> value.replaceAll("\"", "(quote)")}
+     *     <td>{@code { "maskMe": "Andrii (quote)Juice(quote) Pilshchykov" }}
+     * </table>
+     *
+     * <p>Note: usually the {@link ValueMasker} operates on a byte level without parsing JSON values
+     * into intermediate objects. This implementation, however,  needs to allocate a {@link String}
+     * before passing it into the function and then turn it back into a byte array for the replacement,
+     * which introduces some performance overhead.
+     */
+    public static ValueMasker.AnyValueMasker withTextFunction(Function<String, String> masker) {
+        return describe(
+                "withTextFunction (%s)".formatted(masker),
+                context -> {
+                    String decodedValue;
+                    if (context.getByte(0) == '"') {
+                        int encodedIndex = 1; // skip opening quote
+                        int valueEndIndex = context.byteLength() - 1; // skip closing quote
+                        int decodedIndex = 0;
+                        // the length of decodedBytes is guaranteed to be lower or equal to the length
+                        // of the encoded bytes sequence:
+                        // 1. for every encoded character (2 bytes), the output is the character without escape - 1 byte
+                        // 2. for every unicode encoded character (6 bytes), the output character is within 1-3 bytes
+                        // 3. for the pair of unicode encoded surrogates (12 bytes), the output is always 4 bytes
+                        // which means that if any escape is present, the decodedBytes will have some null characters
+                        // at the tail, which are cut from the resulting string
+                        byte[] decodedBytes = new byte[context.byteLength()];
+                        while (encodedIndex < valueEndIndex) {
+                            byte originalByte = context.getByte(encodedIndex++);
+                            // next character is escaped, removing the backslash
+                            if (originalByte == '\\') {
```

Comment:
Also would be nice to switch this if/else to not lose track of context for the else. Alternatively, moving the entire block of code inside the else to a new private method might be nice for readability.